### ISSUE TITLE
Enemy randomizer

### ISSUE THREAD

**Issue by spinshock** [BLCM/BLCMods#616]
There was a way to make the mod work without headhunters...
I can't get it to work... can someone explain to me how to filter it or send me the mod filtered. Please.
I don't know whether to include the:
"#No Headhunters
#/No Headhunters"
If you can explain how to filter this mod step by step I will be really greatful. Thanks in advance.

**Comment by spinshock** [comment]
And also do I have to remove "#" in front of the set lines ?

**Comment by EntityDev** [comment]
Download the UCP Filter Tool. Create an empty text file and instruct the Filter Tool to use it as your default patch. Alternately, you can use the Patch.txt supplied with the Filter Tool. It is the Community Patch. Add a single mod (the Enemy Randomizer). A new node will be created, named "mods". Open it. There are two choices to be made among the Mutually Exclusive nodes; one for Mode (Balanced, Unbalanced, or Screw the Chumps) and then one for either Everything or No Headhunters. Make these choices. Next, you can do one of two things. Either Save the patch and EXEC that file in your game menu, or Export the patch to a new file and EXEC that file instead.

This process comments and uncomments the appropriate lines for you. In other words, it automates the application and removal of the pound sign where it is needed.

You can do all this by hand, but the Filter Tool makes it quick and easy. If you want to do it by hand, comment all the SET statements under the Everything node and uncomment all the SET statements under the No Headhunters node.

Dealing with the Mode, by hand, would be a very tedious process. The randomizer I believe is configured by default to run the Balanced mode. I'd use the Filter Tool if I were you, if you want to run Unbalanced or Screw the Chumps.

Please note: The above reflects my current understanding of how the Enemy Randomizer is constructed. There may be errors in my thinking. Nevertheless, using the Filter Tool will get the job done as described.

Also please note: If I recall correctly, I had some difficulty using a "saved" patch. If the resulting "saved" patch file doesn't work for you, try Exporting the patch to a new file, without structure. EXEC the exported file instead. Again, this is done once you're at the main menu in the game.

**Comment by EntityDev** [comment]
Update:

I continued to have trouble getting the Enemy Randomizer (or any patch, for that matter) with the above instructions.

A slight addition to the above is in order. I have discovered that, at least for me, taking the following extra step is necessary to get the Enemy Randomizer to work consistently.

Once in the game main menu, hit Escape. You will be prompted to return to the Title Screen. Click Yes. Open the console and EXEC the desired patch file. You'll probably want to wait a few seconds. Hit any key to continue and you'll once again be at the main menu. Hit Continue and, if the map takes a long time to load, it's a good sign that the patch was executed and processed.

An additional note about the Filter Tool's behavior when saving a patch. Despite the extra characters it inserts between the Keys and Values SET calls, saved patches appear to work ok.

Good luck..

**Comment by don117** [comment]
It's still not working for me, it loads normally, no errors or anything of this sort, however, when the game loads all I see are the colors from elemental weapons if there are any and a totally black screen - no weapons themselves, no HUD, no nothing. I can jump and hear the jumping sound effect, I can throw a grenade, see it flying and then exploding, but there's no other textures... No nothing, it's like the textures are not loading at all, tried every method I could think off, tried every method I found online, it's still only a black screen after loading the game

**Comment by EntityDev** [comment]
Try selecting the No Headhunters option in Filter Tool. You likely are missing some content.

**Comment by don117** [comment]
Yeah, I've got that out of the way, it seems I was missing UVHU pack 2, now it's all fine

**Comment by tiagonovo96** [comment]
i got a problem too, now is working the mod but i'm invisivle to enemys, they dont see me, why? does anione knows how to fix it?

**Comment by jlpt** [comment]
i checked no headhunters in filter tool and i ran it like how you said it loads but when i try to load a level it loads after like 10 minutes and shows a black screen

**Comment by SgtGatorBite** [comment]
I've been having the black screen issue no matter what setting I change, even if I execute Enemy_Randomizer.txt alone it takes 10 minutes to load and just gives the black screen. You can hear jumping, see grenades being thrown, but can't see anything besides blackness.

**Comment by EntityDev** [comment]
I'm pretty sure it is because you don't have one or more of the maps being loaded. Try the no headhunter's option.

**Comment by EntityDev** [comment]
If that doesn't work, then it is probable that you simply don't have one of the maps that are being loaded;  that your game installation has a corrupt map file or files.

 

You almost assuredly will not receive tangible help from anyone at BLCMods. I have attempted to establish a dialog with one or more of the authors there, but none could put together a coherent sentence. I'm not joking.

 

Quite frankly, running the enemy randomizer in any configuration consumes a huge consumer of machine resources. Besides this - to fully enjoy it, every person who joins your server would also have to have the developer's console enabled, and would have to run the same script. If clients join who are not running the script, they won't be able to see enemies that are from other maps.

 

May I suggest running scripts that concentrate on one map? I create such scripts. All of them contain some features written by others, but the spawn swapping and enemy modifications are mine. I have developed them over a couple of years and we play one of them nearly daily. They are quite challenging, very stable and - not for the faint of heart.

 

They are not necessarily simple to use but, if you've been able to modify your game installation to enable the dev console, then you're probably the type who can figure it out.

 

Give me a holler if you are interested in checking them out, or simply jump in and take a look some time. I'm hosting public nearly every evening.

 

My Steam name changes a lot, depending on which Ultimate script I'm running, but always begins with the string "Zer0 Hour". Look for me.

 

My email address is vb-dev@dispatchmaster.net.
